pkg/hyper/container.go

```diff
@@ -64,3 +85,203 @@ func buildUserContainer(config *kubeapi.ContainerConfig, sandboxConfig *kubeapi.
+
+		if err != nil {
+			glog.Errorf("ParseContainerName for %s failed: %v", c.ContainerName, err)
+			return nil, err
+		}
+
+		if filter != nil {
+			if filter.Id != nil && c.ContainerID != filter.GetId() {
+				continue
+			}
+
+			if filter.PodSandboxId != nil && c.PodID != filter.GetPodSandboxId() {
+				continue
+			}
+
+			if filter.State != nil && state != filter.GetState() {
+				continue
+			}
+		}
+
+		info, err := h.client.GetContainerInfo(c.ContainerID)
+		if err != nil {
+			glog.Errorf("Get container info for %s failed: %v", c.ContainerID, err)
+			return nil, err
+		}
+
+		annotations := getAnnotationsFromLabels(info.Container.Labels)
+		kubeletLabels := getKubeletLabels(info.Container.Labels)
+
+		if filter != nil {
+			if filter.LabelSelector != nil && !inMap(filter.LabelSelector, kubeletLabels) {
+				continue
+			}
+		}
+
+		containerMetadata := &kubeapi.ContainerMetadata{
+			Name:    &containerName,
+			Attempt: &attempt,
+		}
+
+		createdAtNano := info.CreatedAt * secondToNano
+		containers = append(containers, &kubeapi.Container{
+			Id:           &c.ContainerID,
+			PodSandboxId: &c.PodID,
+			CreatedAt:    &createdAtNano,
+			Metadata:     containerMetadata,
+			Image:        &kubeapi.ImageSpec{Image: &info.Container.Image},
+			ImageRef:     &info.Container.ImageID,
+			State:        &state,
+			Labels:       kubeletLabels,
+			Annotations:  annotations,
+		})
+	}
+
+	return containers, nil
+}
+
+// ContainerStatus returns the container status.
+func (h *Runtime) ContainerStatus(containerID string) (*kubeapi.ContainerStatus, error) {
+	status, err := h.client.GetContainerInfo(containerID)
+	if err != nil {
+		glog.Errorf("Get container info for %s failed: %v", containerID, err)
+		return nil, err
+	}
+
+	podInfo, err := h.client.GetPodInfo(status.PodID)
+	if err != nil {
+		glog.Errorf("Get pod info for %s failed: %v", status.PodID, err)
+		return nil, err
+	}
+
+	state := toKubeContainerState(status.Status.Phase)
+	annotations := getAnnotationsFromLabels(status.Container.Labels)
+	kubeletLabels := getKubeletLabels(status.Container.Labels)
+
+	_, _, _, containerName, attempt, err := parseContainerName(strings.Replace(status.Container.Name, "/", "", -1))
+	if err != nil {
+		glog.Errorf("ParseContainerName for %s failed: %v", status.Container.Name, err)
+		return nil, err
+	}
+
+	containerMetadata := &kubeapi.ContainerMetadata{
+		Name:    &containerName,
+		Attempt: &attempt,
+	}
+
+	createdAtNano := status.CreatedAt * secondToNano
+	kubeStatus := &kubeapi.ContainerStatus{
+		Id:          &status.Container.ContainerID,
+		Image:       &kubeapi.ImageSpec{Image: &status.Container.Image},
+		ImageRef:    &status.Container.ImageID,
+		Metadata:    containerMetadata,
+		State:       &state,
+		Labels:      kubeletLabels,
+		Annotations: annotations,
+		CreatedAt:   &createdAtNano,
+	}
+
+	mounts := make([]*kubeapi.Mount, len(status.Container.VolumeMounts))
+	for idx, mnt := range status.Container.VolumeMounts {
```

Comment:
Seems we do not support volumes mounts, but I guess this part does not break anything?